Lovelace: entity-filter: hide card when entities[] empty

ciotlosm opened the pull request:
Implements https://github.com/home-assistant/ui-schema/issues/48

- [ ] developer documentation

Sample:
```yaml
title: Lovelace Demo
views:
  - title: Home
    icon: mdi:heart
    cards:
      - type: entity-filter
        hide_if_empty: true
        filter:
          - domain: lig2ht
        card:
          title: domain light
```

ciotlosm committed:
Implements https://github.com/home-assistant/ui-schema/issues/48

Fixed lint

c727 commented:
maybe we should add this logic to `entities` and `glance` instead, something like `hide if no device in the list is online`?

balloob commented:
I think filter is the right place. A user is explicit in what entities to show for the others, if they don't want to see the card they can remove it

balloob commented on src/panels/lovelace/cards/hui-entity-filter-card.js:
```diff
@@ -77,12 +77,20 @@ class HuiEntitiesCard extends PolymerElement {
   _updateCardConfig(element) {
     if (!element || element.tagName === 'HUI-ERROR-CARD' || !this.hass) return;
+    const entitiesList = this._getEntities(this.hass, this._config.filter);
+    this._computeVisibility(entitiesList.length);
```
I don't really see a reason to extract this into a method

c727 commented:
ok, should we make it optional? `hide_if_empty`

ciotlosm commented:
>ok, should we make it optional? `hide_if_empty`

@c727 it is optional. By default is false

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

balloob commented on src/panels/lovelace/cards/hui-entity-filter-card.js:
```diff
@@ -77,12 +77,20 @@ class HuiEntitiesCard extends PolymerElement {
   }
 
+  _computeVisibility(entitiesLength) {
+    if (this._config.hide_if_empty) {
```
This is incorrect. This means that if I change the config, and set hide_if_empty to false, it might never show itself again.

balloob replied:
I'm also not a big fan of negative config options that then have to be set to true.

I prefer: `show_empty: true/false` with default `true`.

balloob reviewed (changes requested):
We should make sure that changing config resets all things on the element.

ciotlosm committed:
Updates based on review

Reset display on new config

Fixed reset as setConfig was not being called

Fixed lint
